eureka-core/src/main/java/com/netflix/eureka/registry/rule/DownOrStartingRule.java

```diff
@@ -0,0 +1,39 @@
+package com.netflix.eureka.registry.rule;
+
+import com.netflix.appinfo.InstanceInfo;
+import com.netflix.eureka.lease.Lease;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * If the instance is STARTING or DOWN then we return that.
+ * This rule has precedence over the {@link OverrideExistsRule}.
```

Comment:
good point. I updated all the rule comments to be self-contained. Moved descriptions of rule order to the registries.